src/Controller/BulkController.php

```diff
@@ -0,0 +1,249 @@
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function attribute(): ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->request->getData();
+        $this->ids = explode(',', (string)Hash::get($requestData, 'ids'));
+        $this->saveAttribute($requestData['attributes']);
+        $this->errors();
+
+        return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType, '?' => $this->request->getQuery()]);
+    }
+
+    /**
+     * Bulk associate categories to selected ids.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function categories(): ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->request->getData();
+        $this->ids = explode(',', (string)Hash::get($requestData, 'ids'));
+        $this->categories = (string)Hash::get($requestData, 'categoriesSelected');
+        $this->loadCategories();
+        $this->saveCategories();
+        $this->errors();
+
+        return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType, '?' => $this->request->getQuery()]);
+    }
+
+    /**
+     * Bulk associate position in tree to selected ids.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function position(): ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->request->getData();
+        $this->ids = explode(',', (string)Hash::get($requestData, 'ids'));
+        $folder = (string)Hash::get($requestData, 'folderSelected');
+        $action = (string)Hash::get($requestData, 'action');
+        if ($action === 'copy') {
+            $this->copyToPosition($folder);
+        } else { // move
+            $this->moveToPosition($folder);
+        }
+        $this->errors();
+
+        return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType, '?' => $this->request->getQuery()]);
+    }
+
+    /**
+     * Save attribute for selected objects.
+     *
+     * @param array $attributes The attributes data
+     * @return void
+     */
+    protected function saveAttribute(array $attributes): void
+    {
+        foreach ($this->ids as $id) {
+            try {
+                $this->apiClient->save($this->objectType, compact('id') + $attributes);
+            } catch (BEditaClientException $e) {
+                $this->errors[] = ['id' => $id, 'message' => $e->getAttributes()];
+            }
+        }
+    }
+
+    /**
+     * Load categories by type and ids.
+     *
+     * @return void
+     */
+    protected function loadCategories(): void
+    {
+        $url = sprintf('/model/categories?filter[type]=%s&filter[id]=%s', $this->objectType, (string)$this->categories);
+        try {
+            $response = $this->apiClient->get($url, ['page_size' => 100]);
+            $this->categories = (array)Hash::extract($response, 'data.{n}.attributes.name');
```

Comment:
Can we used `categories` already present in `schema` cache? 

For example using 
```php
$schema = $this->Schema->getSchema($this->objectType);
```

Reply:
ok getting categories from schema; but we have to filter them by ids (matching the categories selected by user before submit)